### ISSUE TITLE
Transition exception should show failed validation

### ISSUE THREAD

**Issue by jaylevitt** [pluginaweek/state_machine#142]
I have a state machine like:

```
class AbsenceValidator < ActiveModel::EachValidator
  def validate_each(record, attribute, value)
    unless value.nil?
      record.errors[attribute] << (options[:message] || "must be nil")
    end
  end
end

class SurveyTaking < ActiveRecord::Base
  belongs_to :pending_question

  state_machine :state do
    state :started {validates :pending_question, :presence => true}
    state :completed {validates :pending_question, :absence => true}

    event :complete do
      transition :started => :completed
    end
  end
end

st = SurveyTaking.new
st.pending_question_id = nil
st.valid? #=> false
st.errors #=> {:pending_question=>["must be nil"]}
st.complete! #=> StateMachine::InvalidTransition: Cannot transition state via :complete from :started
```

ActiveRecord (without state_machine) will actually give you the validation in the exception, which is quite helpful:

```
me = User.find(1)
me.gender = 'neither'
me.valid? #=> false
me.save! #=> ActiveRecord::RecordInvalid: Validation failed: Gender is not included in the list
```

I think this was at the root of my #121 issue.

**Comment by obrie** [comment]
Hey @jaylevitt -

I had actually made this change as a fix for #138.  I had admittedly forgotten about #121.  Can you validate that this solution (released in 1.1.0) solves the problem you were seeing?

Thanks.

**Comment by jaylevitt** [comment]
Great, we'll update - thanks!
